src/python/pants/backend/nfpm/target_types.py

```diff
@@ -113,6 +113,13 @@ class NfpmPackageTarget(Target):
     pass
 
 
+# NB: The <packager>_FIELDS constants are used as Nfpm<packager>PackageFieldSet.required_fields,
+# which is used when generating the nfpm config file. The field_set.nfpm_config method loops
+# over required_fields, using the nfpm_alias class var on each field to construct the cpnfig.
+# Some fields, like 'scripts' cannot rely on the nfpm_alias-based logic, however,
+# so they are NOT included in the <packager>_FIELDS constants and are handled separately.
```

Comment:
Not now, but in the future would it make sense to have a `exclude_from_nfpm_alias_logic: ClassVar[bool]` mix-in for the nFPM field classes which would allow the scripts field to declare that it should be excluded from normal processing? Then the scripts field could be included in the various constants because the other parts of the code would know it was special (and future special fields could also declare the opt-out).

Reply:
If `nfpm_alias == ""`, then that is effectively an opt-out for fields like `dependencies`.
https://github.com/pantsbuild/pants/blob/219cb02d43364ca03883d02153011cb66093e852/src/python/pants/backend/nfpm/field_sets.py#L77-L85

The scripts field has `nfpm_alias = ""`:
https://github.com/pantsbuild/pants/blob/219cb02d43364ca03883d02153011cb66093e852/src/python/pants/backend/nfpm/fields/scripts.py#L14-L26

As does `ghost_contents`:
https://github.com/pantsbuild/pants/blob/219cb02d43364ca03883d02153011cb66093e852/src/python/pants/backend/nfpm/fields/rpm.py#L391-L393

So, I suppose we could include them in `<packager>_FIELDS` without issue, and then drop these comments. Would that be cleaner or more consistent?

Reply:
I believe that it would be less confusing for future developers if the constants which purport to be all of the fields are in fact all of the fields. Differentiating based on lack of a nfpm_alias sounds fine. 

For context, I originally asked the question because seeing the scripts fields excluded from the constants made me question why they were excluded. Moreover, it meant that the `FOO_FIELDS` constant was not in fact all of `FOO`'s fields. 

So the thoughts are either include scripts fields in the constants and exclude them in the moment based on the absence of the nfpm_alias as you suggest or devise better naming for the constants to reflect they are not all of the fields.

Reply:
I think moving the scripts fields into the constants is the cleanest approach, so I'll do that. Thanks for the feedback